src/linker/Linker/MessageContainer.cs

```diff
@@ -70,6 +70,9 @@ public static MessageContainer CreateWarningMessage (LinkContext context, string
 			if (context.IsWarningSuppressed (code, origin))
 				return Empty;
 
+			if (context.WarningSuppressionWriter != null && origin.MemberDefinition != null)
```

Comment:
I think this is an incorrect place, the act of creating the message instance does not mean it will be sent to the output